packages/opentelemetry-metrics/src/MetricObservable.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as api from '@opentelemetry/api';
+
+type Subscriber = (value?: number) => void;
+
+/**
+ * Implements the Metric Observable pattern
+ */
+export class MetricObservable implements api.MetricObservable {
+  private _subscribers: Subscriber[] = [];
+
+  next(value: number) {
+    this._subscribers.forEach(subscriber => {
```

Comment:
Using `for..of` would avoid creating this arrow function in memory.

Reply:
We should probably also wrap each call in a `try/catch`

Reply:
with regards to try catch - I would argue if this is a good idea, I will add extra overhead and will fail silently instead. I would rather let any failures to be seen faster

Reply:
ok